Helper/Data.php

```diff
@@ -129,14 +144,18 @@ public function setProgressFile(
         $isUploaded = "false",
         $error = ""
     ) {
-        if ($progressFileName != null) {
-            $progressFile = fopen($progressFileName, "w");
-            fwrite(
-                $progressFile,
-                "{\"name\":\"$feedName\",\"cur\":$currentPage,\"max\":$pages,\"isComplete\":$isComplete,"
-                . "\"isUploaded\":$isUploaded,\"error\":\"$error\"}"
-            );
-            fclose($progressFile);
+        if ($progressFileName !== null) {
+            try {
+                $progressFile = $this->driver->fileOpen($progressFileName, "w");
+                $this->driver->fileWrite(
+                    $progressFile,
+                    "{\"name\":\"$feedName\",\"cur\":$currentPage,\"max\":$pages,\"isComplete\":$isComplete,"
+                    . "\"isUploaded\":$isUploaded,\"error\":\"$error\"}"
+                );
+                $this->driver->fileClose($progressFile);
```

Comment:
might have opened, then get exception in the write. File would be left open?

Reply:
So in this try > catch, there are 3 points that an exception can be thrown:

1. on opening the file
2. on writing to the file
3. on closing the file.

I think, if it happens on 2, the file will still technically be "open", so I'm wondering if i need to split these out into 3 try - catches instead, one for each of the points above, and if it catches on 2, then i close the file (won't need to close it on 1 / 3 as wither the file won't be open, or we can't actually close the file)